lib/pbench/server/__init__.py

```diff
@@ -38,135 +38,144 @@ class OperationCode(Enum):
+    )
+
     # Define a fallback default for dataset maximum retention, which we expect
     # to be defined in pbench-server-default.cfg.
     MAXIMUM_RETENTION_DAYS = 3650
 
-    def __init__(self, cfg_name):
+    def __init__(self, cfg_name: str):
+        """Constructor to add specific fields for the operation of the server.
+
+        Args:
+            cfg_name : The configuration file name to use
+        """
         super().__init__(cfg_name)
 
-        # Now fetch some default common pbench settings that are required.
-        self.TOP = self._get_valid_dir_option("TOP", "pbench-server", "pbench-top-dir")
-        self.TMP = self._get_valid_dir_option("TMP", "pbench-server", "pbench-tmp-dir")
-        if self.log_dir:
-            # We have a logging directory, which means the logger_type is
-            # "file", so we'll ignore any "pbench-logs-dir" configuration
-            # values.
-            self.LOGSDIR = Path(self.log_dir)
-        else:
-            # We don't have a [logging] section "log_dir" option, so we'll
-            # fetch the old "pbench-logs-dir" option.
-            self.LOGSDIR = self._get_valid_dir_option(
-                "LOGSDIR", "pbench-server", "pbench-logs-dir"
-            )
-            # Provide a value for log_dir since it was provided via the old
-            # pbench-logs-dir configuration.
-            self.log_dir = str(self.LOGSDIR)
         # The pbench server, unlike the pbench agent code, logs to separate
         # directories for each caller.
         self.log_using_caller_directory = True
-        self.BINDIR = self._get_valid_dir_option(
-            "BINDIR", "pbench-server", "script-dir"
-        )
-        self.LIBDIR = self._get_valid_dir_option("LIBDIR", "pbench-server", "lib-dir")
-        self.ARCHIVE = self._get_valid_dir_option(
-            "ARCHIVE", "pbench-server", "pbench-archive-dir"
-        )
 
-        self.INCOMING = self.TOP / "public_html" / "incoming"
-        # this is where the symlink forest is going to go
-        self.RESULTS = self.TOP / "public_html" / "results"
-        self.USERS = self.TOP / "public_html" / "users"
+        # Constants
 
-        try:
-            self.PBENCH_ENV = self.conf.get("pbench-server", "environment")
-        except NoOptionError:
-            self.PBENCH_ENV = ""
+        # Initial common timestamp format
+        self.TS = f"run-{tstos(_time())}"
 
-        try:
-            self.COMMIT_ID = self.conf.get("pbench-server", "commit_id")
-        except NoOptionError:
-            self.COMMIT_ID = "(unknown)"
+    @property
+    def TOP(self) -> Path:
+        return self._get_valid_dir_option("TOP", "pbench-server", "pbench-top-dir")
 
-        if self._unittests:
+    @property
+    def TMP(self) -> Path:
+        return self._get_valid_dir_option("TMP", "pbench-server", "pbench-tmp-dir")
 
-            def mocked_time():
-                return 42.00
+    @property
+    def LOGSDIR(self) -> Path:
+        if self.log_dir:
```

Comment:
We should initialize `log_dir` in `__init__()`.  (How are we avoiding an `AttributeError` on the first call here??)

Reply:
The value is always set by the base class, see line 45 of `lib/pbench/__init__.py`.

Reply:
OK, good.  However, this leaves me wondering whether this method shouldn't be hoisted to the base class....

Reply:
Great idea for another PR to consider such a change.  Thanks!